packages/ruleset/src/functions/inline-response-schema.js

```diff
@@ -0,0 +1,68 @@
+const each = require('lodash/each');
+const {
+  walk,
+  isJsonMimeType,
+  isPrimitiveType,
+  isResponseObject
+} = require('../utils');
+
+const errorMessage =
+  'Response schemas should be defined as a $ref to a named schema.';
+
+/**
+ * Checks to make sure that response schemas are defined as a ref to a named schema,
+ * rather than an inline schema.
+ * Even if the referenced schema is only used once, naming it offers significant benefits for SDK generation,
+ * because the SDK generator will use the name of the referenced schema as the operation's return type, and
+ * allowing the generator to perform its automatic refactoring will result in sub-optimal names.
+ * @param {*} apidef the entire (un-resolved) API definition
+ * @returns an array containing the violations found or [] if no violations
+ */
+function inlineResponseSchema(apidef) {
+  const errors = [];
+
+  // Walk the entire API definition
+  walk(apidef, [], function(obj, path) {
```

Comment:
I thought about doing that before, but chickened out in favor of trying to port the legacy code forward as-is , but I agree it's better to avoid using "walk".   I'll fix that.

Reply:
I ended up changing the rule definition and the function so it is called directly for each response schema within the unresolved API definition.  This ended up simplifying the rule function by a lot.